otel.go

```diff
@@ -20,30 +21,38 @@ const (
 	packageName = "github.com/evergreen-ci/logkeeper"
 )
 
-func (lk *logkeeper) initOtel(ctx context.Context) error {
-	if lk.opts.TraceCollectorEndpoint == "" {
+var closers []closerOp
+
+func InitOtel(ctx context.Context, collectorEndpoint string) (otelTrace.Tracer, error) {
+	var (
+		err          error
+		otelGrpcConn *grpc.ClientConn
+		tracer       = otel.GetTracerProvider().Tracer("noop_tracer") // default
+		r            *resource.Resource
+	)
+	if collectorEndpoint == "" {
 		// defaults to NoopTracerProvider
-		lk.tracer = otel.GetTracerProvider().Tracer(packageName)
-		return nil
+		tracer = otel.GetTracerProvider().Tracer(packageName)
+		return tracer, nil
 	}
 
-	r, err := serviceResource(ctx)
+	r, err = serviceResource(ctx)
 	if err != nil {
-		return errors.Wrap(err, "making host resource")
+		return tracer, errors.Wrap(err, "making host resource")
 	}
 
-	lk.otelGrpcConn, err = grpc.DialContext(ctx,
-		lk.opts.TraceCollectorEndpoint,
+	otelGrpcConn, err = grpc.DialContext(ctx,
+		collectorEndpoint,
 		grpc.WithTransportCredentials(credentials.NewTLS(nil)),
 	)
```

Comment:
if we don't need to hold on to the connection we could allow the otlptracegrpc.Client to manage its own connection. (like we do [here](https://github.com/evergreen-ci/evergreen/blob/352b79297bf6aa298879c37a98b93525d3311dcd/environment.go#L901))